src/channel_manager.ts

```diff
@@ -0,0 +1,517 @@
+import type { StreamChat } from './client';
+import type {
+  DefaultGenerics,
+  ExtendableGenerics,
+  Event,
+  ChannelOptions,
+  ChannelStateOptions,
+  ChannelFilters,
+  ChannelSort,
+} from './types';
+import { StateStore } from './store';
+import { Channel } from './channel';
+import {
+  findLastPinnedChannelIndex,
+  findPinnedAtSortOrder,
+  getAndWatchChannel,
+  isChannelArchived,
+  isChannelPinned,
+  moveChannelUpwards,
+  shouldConsiderArchivedChannels,
+  shouldConsiderPinnedChannels,
+} from './utils';
+
+export type ChannelManagerPagination<SCG extends ExtendableGenerics = DefaultGenerics> = {
+  filters: ChannelFilters<SCG>;
+  hasNext: boolean;
+  isLoading: boolean;
+  isLoadingNext: boolean;
+  options: ChannelOptions;
+  sort: ChannelSort<SCG>;
+  stateOptions: ChannelStateOptions;
```

Comment:
I am wondering, if one day we would introduce configuration parameters that would not be related to state neither to HTTP request. In that case we would probably need another XOptions field?

Another thing I am thinking - do the `skipInitialization` and `offlineMode` need to me reactive?

Reply:
Answered [here](https://github.com/GetStream/stream-chat-js/pull/1460/files#r1939379422), will do that - thanks !